waiter/src/waiter/auth/authentication.clj

```diff
@@ -28,21 +29,30 @@
     "Attaches middleware that enables the application to perform authentication.
      The middleware should
      - either issue a 401 challenge asking the client to authenticate itself,
-     - or upon successful authentication populate the request with :authorization/user and :authenticated-principal"))
+     - or upon successful authentication populate the request with :authorization/user and :authorization/principal"))
 
 (defn- add-cached-auth
   [response password principal]
   (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (System/currentTimeMillis)] 1))
 
+(defn assoc-auth-params
+  "Associate values for authenticated user in the request."
+  ([request principal]
+   (assoc-auth-params request (first (str/split principal #"@" 2)) principal))
+  ([request user principal]
```

Comment:
Nitpick: we should make the new argument (`user`) the last in the overloaded version in this scenario.